### ISSUE TITLE
SOSP 19 | A Generic Communication Scheduler for Distributed DNN Training Acceleration

### ISSUE THREAD

**Issue by ANormalMan12** [AI-Infra-Team/awesome-papers#2]
A generic communication scheduler for distributed DNN training acceleration.

Authors: Yanghua Peng, Yibo Zhu, Yangrui Chen, Yixin Bao, Bairen Yi, Chang Lan,  Chuan Wu, Chuanxiong Guo

Motivation: The paper found that existing DNN training frameworks inefficiently utilize the bandwidth of the network and introduce unnecessary latency because they use FIFO methods to schedule communication tasks.

Method: 
1. A theoretical optimal strategy that overlaps the communication of iteration i and the computation of iteration i+1 by assigning higher priority to layers near DNN inputs and partitioning the tensors to send. 
2. A unified abstraction that integrates the scheduler inside multiple deep learning framework (MXNet, PyTorch, TensorFlow) without modifying their upper-level APIs or lower-level execution engines.
3. An auto-tuning module based on Bayesian Optimization to determine priority of tensors and the partition size of tensors under diverse runtime conditions, such as varying network bandwidths and model architectures.

**Comment by qzweng** [comment]
released code: https://github.com/bytedance/byteps
related paper: OSDI 20 | A Unified Architecture for Accelerating Distributed DNN Training in Heterogeneous GPU/CPU Clusters [link](https://www.usenix.org/conference/osdi20/presentation/jiang)
